packages/core/src/components/grid/grid.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react'
+import type { TLGrid } from '~types'
+
+export function Grid({
+  id,
+  grid,
+  camera,
+}: {
+  id: string
+  grid: TLGrid
+  camera: { point: number[]; zoom: number }
+}) {
+  const s = grid.size * camera.zoom
+  const xo = camera.point[0] * camera.zoom
+  const yo = camera.point[1] * camera.zoom
+  const gxo = xo > 0 ? xo % s : s + (xo % s)
+  const gyo = yo > 0 ? yo % s : s + (yo % s)
+
+  return (
+    <>
+      <defs>
+        <pattern id={id} width={s} height={s} patternUnits="userSpaceOnUse">
+          <line x1={gxo} y1={0} x2={gxo} y2={s} fill="none" stroke={grid.color} strokeWidth={1} />
+          <line x1={0} y1={gyo} x2={s} y2={gyo} fill="none" stroke={grid.color} strokeWidth={1} />
```

Comment:
This could perhaps be simplify it by using a single `path` instead of `line` here, that way we can avoid having to update multiple DOM nodes on each camera update.